src/components/ScreenWrapper.js

```diff
@@ -116,4 +129,11 @@ class ScreenWrapper extends React.Component {
 ScreenWrapper.propTypes = propTypes;
 ScreenWrapper.defaultProps = defaultProps;
 ScreenWrapper.displayName = 'ScreenWrapper';
-export default withNavigation(ScreenWrapper);
+export default compose(
+    withNavigation,
+    withOnyx({
+        modal: {
```

Comment:
NAB: Why does the modal state need to be in Onyx? Does't the modal's visibility only really matter to the client?

Reply:
Yup. You are correct but App is architected this way. and we only use Onyx for data flow.